src/buildConfig.js

```diff
@@ -44,7 +46,7 @@ export default function (options) {
   const configBasedir = options.configBasedir || false
 
   if (rawConfig) {
-    const configDir = configBasedir || process.cwd()
+    const configDir = configBasedir || (options.filepath ? path.dirname(options.filepath) : process.cwd())
```

Comment:
In Atom `linter-stylelint` and some third party utilities that run `stylelint` will pass a `cosmiconfig` result object to `stylelint`. result.filepath is configuration file's path

If a `options` is `cosmiconfig` result object, we shold do logic same with this part of code:

```
  return cosmiconfig("stylelint", cosmiconfigOptions).then(result => {
    if (!result) throw configurationError("No configuration found")
    rootConfigDir = configBasedir || path.dirname(result.filepath)
    return augmentConfig(result.config, rootConfigDir, {
      addIgnorePatterns: true,
      ignorePath: options.ignorePath,
    })
  })
```
`rootConfigDir = configBasedir || path.dirname(result.filepath)`